### PR TITLE
[new release] postgresql (5.1.0)

### PR DESCRIPTION
**Pull request by mmottl**
Bindings to the PostgreSQL library

- Project page: <a href="https://mmottl.github.io/postgresql-ocaml">https://mmottl.github.io/postgresql-ocaml</a>
- Documentation: <a href="https://mmottl.github.io/postgresql-ocaml/api">https://mmottl.github.io/postgresql-ocaml/api</a>

##### CHANGES:

- Made Postgresql.null a now unique, empty string.

- Fixed license format and a typo.

- Fixed some odoc references.

- Added GitHub workflows.

- Added automatic formatting with ocamlformat and clang-format.

- Ported the config discovery script to pkg-config

  Thanks to Antonio Nuno Monteiro for the patch.

- Improved Dune rules.

- Used new OCaml 4.12 C-macros.

- Switched to Dune lang 2.7.

### CODE REVIEW

**Comment by mmottl** [comment]
I hope this new release will work fine with the transition to `pkg-config`. It is likely that `conf-postgresql` will need to be adjusted for different platforms, but it's a start.

Here are more details about the change: https://github.com/mmottl/postgresql-ocaml/pull/48

File: packages/conf-mingw-w64-postgresql-i686/conf-mingw-w64-postgresql-i686.1/opam [hunk 1]
```diff
@@ -0,0 +1,19 @@
+opam-version: "2.0"
+synopsis: "postgresql for i686 mingw-w64 (32-bit x86)"
```
Comment (mseri) [inline packages/conf-mingw-w64-postgresql-i686/conf-mingw-w64-postgresql-i686.1/opam:2]:
```suggestion
synopsis: "Postgresql for i686 mingw-w64 (32-bit x86)"
```

File: packages/conf-mingw-w64-postgresql-x86_64/conf-mingw-w64-postgresql-x86_64.1/opam [hunk 2]
```diff
@@ -0,0 +1,19 @@
+opam-version: "2.0"
+synopsis: "postgresql for x86_64 mingw-w64 (64-bit x86_64)"
```
Comment (mseri) [inline packages/conf-mingw-w64-postgresql-x86_64/conf-mingw-w64-postgresql-x86_64.1/opam:2]:
```suggestion
synopsis: "Postgresql for x86_64 mingw-w64 (64-bit x86_64)"
```

**Comment by mseri** [comment]
The failure on macos is to be expected unfortunately
```
- libpq is keg-only, which means it was not symlinked into /opt/homebrew,
- because conflicts with postgres formula.
- 
- If you need to have libpq first in your PATH, run:
-   echo 'export PATH="/opt/homebrew/opt/libpq/bin:$PATH"' >> ~/.profile
- 
- For compilers to find libpq you may need to set:
-   export LDFLAGS="-L/opt/homebrew/opt/libpq/lib"
-   export CPPFLAGS="-I/opt/homebrew/opt/libpq/include"
- 
- For pkg-config to find libpq you may need to set:
-   export PKG_CONFIG_PATH="/opt/homebrew/opt/libpq/lib/pkgconfig"
```
but I don't think we can do better

**Comment by avsm** [comment]
Thanks! You may want to announce this on <https://discuss.ocaml.org>, where we have a Community category and an `announce` tag for this purpose.

**Pull request merged**